cloudferrylib/os/image/glance_image.py

```diff
@@ -214,8 +215,26 @@ def deploy(self, info, callback=None):
         return new_info
 
     def wait_for_status(self, id_res, status):
-        while self.glance_client.images.get(id_res).status != status:
-            time.sleep(1)
+        limit_retry = self.config.image.wait_for_status_retries
+        retry_interval = self.config.image.wait_for_status_interval
+        if limit_retry <= 0:
+            LOG.warn("Treating negative or zero config value %s "
+                     "for 'wait_for_status_retries' of image service."
+                     % limit_retry)
+            limit_retry = 60
+        if retry_interval <= 0:
+            LOG.warn("Treating negative or zero config value %s "
+                     "for 'wait_for_status_interval' of image service."
+                     % retry_interval)
+            retry_interval = 3
```

Comment:
Retry logic is identical for any task. Lets move it to a separate module and use it from the tasks in question.